src/metabase/driver/common/parameters/dates.clj

```diff
@@ -199,6 +219,46 @@
   [{:keys [start end]} field-clause]
   [:between (mbql.u/with-temporal-unit field-clause :day) (->iso-8601-date start) (->iso-8601-date end)])
 
+(def ^:private short-day->day
+  {"Mon" :monday
+   "Tue" :tuesday
+   "Wed" :wednesday
+   "Thu" :thursday
+   "Fri" :friday
+   "Sat" :saturday
+   "Sun" :sunday})
+
+(def ^:private short-month->month
+  (into {}
+        (map-indexed (fn [i m] [m (inc i)]))
+        ["Jan" "Feb" "Mar" "Apr" "May" "Jun" "Jul" "Aug" "Sep" "Oct" "Nov" "Dec"]))
+
+(defn- parse-int-in-range [s min-val max-val]
+  (try
+    (let [i (Integer/parseInt s)]
+      (when (<= min-val i max-val)
+        i))
+    (catch NumberFormatException _)))
+
+(defn- excluded-datetime [unit date exclusion]
+  (let [year (t/year date)]
+    (case unit
+      :hour (when-let [hour (parse-int-in-range exclusion 0 23)]
+              (format "%sT%02d:00:00Z" date hour))
+      :day (when-let [day (short-day->day exclusion)]
+             (str (t/adjust date :next-or-same-day-of-week day)))
+      :month (when-let [month (short-month->month exclusion)]
+               (format "%s-%02d-01" year month))
+      :quarter (when-let [quarter (parse-int-in-range exclusion 1 4)]
+                 (format "%s-%02d-01" year (inc (* 3 (dec quarter)))))
+      nil)))
```

Comment:
Do you think this case statement is a good place to throw if it gets passed an invalid unit? If so I'd remove this `nil`

Reply:
`nil` is used signal that this parser cannot parse the input.